In [1]:
import pandas as pd
import os

In [ ]:
data = pd.read_csv(os.getcwd()+"\\dataset\\whisky_basic\\save_1_600.csv", index_col = 0)

In [2]:
meta_data = pd.read_csv(os.getcwd()+"\\dataset\\whisky_metadata\\metadata_0_700.csv", index_col = 0)

In [3]:
data

link  \
0                               /spirits/hibiki-21-year   
1                             /spirits/highland-park-18   
2     /spirits/michter-s-20-year-kentucky-straight-b...   
3             /spirits/george-t-stagg-bourbon-fall-2019   
4                 /spirits/bowmore-mizunara-cask-finish   
...                                                 ...   
3652              /spirits/westland-single-cask-no-1766   
3653  /spirits/barrell-whiskey-infinite-barrel-proje...   
3654       /spirits/fercullen-single-malt-aged-20-years   
3655  /spirits/knob-creek-single-barrel-liquor-barn-...   
3656       /spirits/knob-creek-single-barrel-fred-noe-6   

                                                  image  \
0     https://ip-distiller.imgix.net/images/spirits/...   
1     https://ip-distiller.imgix.net/images/spirits/...   
2     https://ip-distiller.imgix.net/images/spirits/...   
3     https://ip-distiller.imgix.net/images/spirits/...   
4     https://ip-distiller.imgix.net/images/spirits/...   
...                                                 ...   
3652  https://ip-distiller.imgix.net/images/spirits/...   
3653  https://ip-distiller.imgix.net/images/spirits/...   
3654  https://ip-distiller.imgix.net/images/spirits/...   
3655  https://ip-distiller.imgix.net/images/spirits/...   
3656  https://ip-distiller.imgix.net/images/spirits/...   

                                                   name  avr_rating  \
0                                        Hibiki 21 Year        4.52   
1                                 Highland Park 18 Year        4.47   
2     Michter's 20 Year Kentucky Straight Bourbon (2...        4.61   
3                   George T. Stagg Bourbon (Fall 2019)        4.61   
4                          Bowmore Mizunara Cask Finish        4.08   
...                                                 ...         ...   
3652                      Westland Single Cask No. 1766        5.00   
3653  Barrell Whiskey Infinite Barrel Project (Dec, ...         NaN   
3654                Fercullen Single Malt aged 20 years         NaN   
3655     Knob Creek Single Barrel "Liquor Barn 2021 #6"        4.00   
3656             Knob Creek Single Barrel "Fred Noe #6"        4.00   

                  category           location  
0                  Blended              Japan  
1       Peated Single Malt  Islands, Scotland  
2                  Bourbon      Kentucky, USA  
3                  Bourbon      Kentucky, USA  
4       Peated Single Malt    Islay, Scotland  
...                    ...                ...  
3652  American Single Malt                USA  
3653               Bourbon                USA  
3654           Single Malt            Ireland  
3655               Bourbon                USA  
3656               Bourbon                USA  

[3657 rows x 6 columns]

In [4]:
meta_data

name  total_rating  \
0                                       Hibiki 21 Year           861   
1                                Highland Park 18 Year          2987   
2    Michter's 20 Year Kentucky Straight Bourbon (2...            10   
3                  George T. Stagg Bourbon (Fall 2019)           628   
4                         Bowmore Mizunara Cask Finish            22   
..                                                 ...           ...   
695                             Port Charlotte 10 Year          1163   
696          Compass Box Flaming Heart (Sixth Edition)            50   
697     Booker's Bourbon Batch 2018-03 "Kentucky Chew"           199   
698           Highland Park Wings of the Eagle 16 Year            50   
699  Old Fitzgerald Bottled in Bond 9 Year (Fall 2018)           174   

     cost_rank    abv                                          cask_type  \
0            5  43.00  ex-bourbon American oak, ex-sherry European oa...   
1            4  43.00                                          ex-sherry   
2            5  57.10                          new, charred American oak   
3            4  58.45                          new, charred American oak   
4            5  53.90            ex-bourbon, ex-sherry, new Mizunara oak   
..         ...    ...                                                ...   
695          3  50.00  1st and 2nd fill American whiskey casks & 2nd ...   
696          4  48.90  a variety of casks including heavily-toasted n...   
697          3  63.35                          new, charred American oak   
698          4  44.50                   primarily European oak ex-sherry   
699          4  50.00                          new, charred American oak   

     smoky  peaty  spicy  herbal  oily  full_bodied  rich  sweet  briny  \
0       30     85     50      30    20           80    80     85      0   
1       40     30     40      20    40           70    80     70     20   
2       15      0     20       0    15           80    90     85      0   
3       40      0     65      50    20           60    60     45      0   
4       30     20     40      30    10           75    75     60     15   
..     ...    ...    ...     ...   ...          ...   ...    ...    ...   
695     60     70     40      25    55           60    65     35     35   
696     80     30     50       0    10           80    70     60     30   
697     35      0     65      25    60           60    70     60      5   
698     50     60     40      40    30           70    80     60      0   
699     25      0     55      35    25           50    50     55      0   

     salty  vanilla  tart  fruity  floral  
0       15       20    25      85      50  
1       40       50    50      70      20  
2        5       30    25      35       0  
3        0       60    60      45       0  
4       20       30    20      50       5  
..     ...      ...   ...     ...     ...  
695     15       35    15      20      15  
696     20       70    10      50       0  
697      5       55    10      15      10  
698      0       80     0      50      30  
699     10       35    40      50       0  

[700 rows x 19 columns]

In [5]:
meta_700 = pd.read_csv(os.getcwd()+"\\dataset\\whisky_metadata\\metadata_700_1400.csv", index_col=0)
meta_1400 = pd.read_csv(os.getcwd()+"\\dataset\\whisky_metadata\\metadata_1400_2100.csv", index_col=0)
meta_2100 = pd.read_csv(os.getcwd()+"\\dataset\\whisky_metadata\\metadata_2100_2800.csv", index_col=0)
meta_2800 = pd.read_csv(os.getcwd()+"\\dataset\\whisky_metadata\\metadata_2800_3535.csv", index_col=0)

In [6]:
meta_data = pd.concat([meta_data, meta_700, meta_1400, meta_2100, meta_2800])

In [7]:
meta_data

name  total_rating  \
0                                        Hibiki 21 Year           861   
1                                 Highland Park 18 Year          2987   
2     Michter's 20 Year Kentucky Straight Bourbon (2...            10   
3                   George T. Stagg Bourbon (Fall 2019)           628   
4                          Bowmore Mizunara Cask Finish            22   
...                                                 ...           ...   
3530                                 Wild Turkey Spiced            13   
3531         Seagram's 7 Crown American Blended Whiskey           237   
3532                       11 Wells Single Malt Whiskey             2   
3533                     Immortal Spirits Early Whiskey             1   
3534                       Triple Crown Blended Whiskey            16   

      cost_rank    abv                                          cask_type  \
0             5  43.00  ex-bourbon American oak, ex-sherry European oa...   
1             4  43.00                                          ex-sherry   
2             5  57.10                          new, charred American oak   
3             4  58.45                          new, charred American oak   
4             5  53.90            ex-bourbon, ex-sherry, new Mizunara oak   
...         ...    ...                                                ...   
3530          1  43.00                          New, Charred American Oak   
3531          1  40.00                                                oak   
3532          3  42.00                          new, charred American oak   
3533          2  44.50                          new, charred American oak   
3534          1  40.00                                                NaN   

      smoky  peaty  spicy  herbal  oily  full_bodied  rich  sweet  briny  \
0        30     85     50      30    20           80    80     85      0   
1        40     30     40      20    40           70    80     70     20   
2        15      0     20       0    15           80    90     85      0   
3        40      0     65      50    20           60    60     45      0   
4        30     20     40      30    10           75    75     60     15   
...     ...    ...    ...     ...   ...          ...   ...    ...    ...   
3530     35      0     20       0     0           40    55     85      0   
3531      0      0     40      20     0           30     0    100      0   
3532     90     10     10      10    40           10    10     90      0   
3533      0      0     10     100    80            0     0     40      0   
3534     31      0     40      29     0           30    40    100      0   

      salty  vanilla  tart  fruity  floral  
0        15       20    25      85      50  
1        40       50    50      70      20  
2         5       30    25      35       0  
3         0       60    60      45       0  
4        20       30    20      50       5  
...     ...      ...   ...     ...     ...  
3530      0       25    10      10       0  
3531      0      100     0      60      10  
3532      0       50    20      30      10  
3533      0       20     0       0       0  
3534      0       55     0      30      10  

[3535 rows x 19 columns]

### total review count = 907948

In [8]:
meta_data['total_rating'].sum()

907948

### Meta Data Summary

In [9]:
meta_data.describe()

total_rating    cost_rank          abv        smoky        peaty  \
count   3535.000000  3535.000000  3532.000000  3535.000000  3535.000000   
mean     256.845262     3.141160    47.507081    22.308911     8.764356   
std      821.089471     1.137736     6.462823    20.543387    19.420631   
min        0.000000     1.000000     0.000000     0.000000     0.000000   
25%       12.000000     2.000000    43.000000     5.000000     0.000000   
50%       44.000000     3.000000    46.000000    20.000000     0.000000   
75%      152.000000     4.000000    50.000000    30.000000     0.000000   
max    12255.000000     5.000000    80.000000   100.000000   100.000000   

             spicy       herbal         oily  full_bodied         rich  \
count  3535.000000  3535.000000  3535.000000  3535.000000  3535.000000   
mean     45.916549    26.206789    28.588119    54.667327    54.431966   
std      20.427383    19.450020    19.680687    18.616834    19.170765   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%      30.000000    10.000000    15.000000    40.000000    40.000000   
50%      45.000000    20.000000    25.000000    55.000000    55.000000   
75%      60.000000    40.000000    40.000000    70.000000    70.000000   
max     100.000000   100.000000   100.000000   100.000000   100.000000   

             sweet        briny        salty      vanilla         tart  \
count  3535.000000  3535.000000  3535.000000  3535.000000  3535.000000   
mean     54.382461     8.150495     9.555304    42.981895    25.253748   
std      18.389508    14.727173    14.813603    21.096638    18.311131   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%      40.000000     0.000000     0.000000    25.000000    10.000000   
50%      60.000000     0.000000     0.000000    40.000000    20.000000   
75%      70.000000    10.000000    15.000000    60.000000    40.000000   
max     100.000000   100.000000    90.000000   100.000000   100.000000   

            fruity       floral  
count  3535.000000  3535.000000  
mean     44.203112    20.699010  
std      22.691156    20.244514  
min       0.000000     0.000000  
25%      25.000000     4.000000  
50%      40.000000    15.000000  
75%      60.000000    30.000000  
max     100.000000    95.000000

In [10]:
meta_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3535 entries, 0 to 3534
Data columns (total 19 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   name          3535 non-null   object 
 1   total_rating  3535 non-null   int64  
 2   cost_rank     3535 non-null   int64  
 3   abv           3532 non-null   float64
 4   cask_type     2981 non-null   object 
 5   smoky         3535 non-null   int64  
 6   peaty         3535 non-null   int64  
 7   spicy         3535 non-null   int64  
 8   herbal        3535 non-null   int64  
 9   oily          3535 non-null   int64  
 10  full_bodied   3535 non-null   int64  
 11  rich          3535 non-null   int64  
 12  sweet         3535 non-null   int64  
 13  briny         3535 non-null   int64  
 14  salty         3535 non-null   int64  
 15  vanilla       3535 non-null   int64  
 16  tart          3535 non-null   int64  
 17  fruity        3535 non-null   int64  
 18  floral        3535 non-null 

In [11]:
meta_data.loc[meta_data["abv"].isnull()]

name  total_rating  cost_rank  \
1686    WhistlePig The Boss Hog IX: Siren's Song            11          5   
2987                 Onyx Moonshine Secret Stash             6          3   
3172  Double Diamond Limited Edition Bourbon 267             1          2   

      abv                                          cask_type  smoky  peaty  \
1686  NaN  American oak; finished in Greek fig nectar and...     40      0   
2987  NaN  5 gallon new, charred American oak sealed in o...     10      0   
3172  NaN                                       American Oak      0      0   

      spicy  herbal  oily  full_bodied  rich  sweet  briny  salty  vanilla  \
1686     85      25    20           65    65     45      0      0       20   
2987     20       0    30           50    40     30      0      0       30   
3172     30       0    30           20    20     40      0      0       50   

      tart  fruity  floral  
1686    25      55      25  
2987    10      10       0  
3172     0      10      20

1686 => 51.7

### 도수가 null인 위스키 3개 
--- 
- 직접 검색하여 입력해줬습니다.

In [12]:
meta_data.loc[1686, 'abv'] = 51.7
meta_data.loc[2987, 'abv'] = 45.00
meta_data.loc[3172, 'abv'] = 40.00

## Whisky 기본 Data + Meta Data 작업 수행

0~3534 까지의 위스키 정보 사용
- 즉, 3535개의 Unique한 위스키가 존재합니다.

In [13]:
processed_data = pd.merge(data, meta_data)
processed_data

link  \
0                               /spirits/hibiki-21-year   
1                             /spirits/highland-park-18   
2     /spirits/michter-s-20-year-kentucky-straight-b...   
3             /spirits/george-t-stagg-bourbon-fall-2019   
4                 /spirits/bowmore-mizunara-cask-finish   
...                                                 ...   
3530                        /spirits/wild-turkey-spiced   
3531  /spirits/seagram-s-seven-crown-american-blende...   
3532              /spirits/11-wells-single-malt-whiskey   
3533            /spirits/immortal-spirits-early-whiskey   
3534              /spirits/triple-crown-blended-whiskey   

                                                  image  \
0     https://ip-distiller.imgix.net/images/spirits/...   
1     https://ip-distiller.imgix.net/images/spirits/...   
2     https://ip-distiller.imgix.net/images/spirits/...   
3     https://ip-distiller.imgix.net/images/spirits/...   
4     https://ip-distiller.imgix.net/images/spirits/...   
...                                                 ...   
3530  https://ip-distiller.imgix.net/images/spirits/...   
3531  https://ip-distiller.imgix.net/images/spirits/...   
3532  https://ip-distiller.imgix.net/images/spirits/...   
3533  https://ip-distiller.imgix.net/images/spirits/...   
3534  https://ip-distiller.imgix.net/images/spirits/...   

                                                   name  avr_rating  \
0                                        Hibiki 21 Year        4.52   
1                                 Highland Park 18 Year        4.47   
2     Michter's 20 Year Kentucky Straight Bourbon (2...        4.61   
3                   George T. Stagg Bourbon (Fall 2019)        4.61   
4                          Bowmore Mizunara Cask Finish        4.08   
...                                                 ...         ...   
3530                                 Wild Turkey Spiced        3.00   
3531         Seagram's 7 Crown American Blended Whiskey        2.21   
3532                       11 Wells Single Malt Whiskey        3.00   
3533                     Immortal Spirits Early Whiskey        1.00   
3534                       Triple Crown Blended Whiskey        2.43   

                      category           location  total_rating  cost_rank  \
0                      Blended              Japan           861          5   
1           Peated Single Malt  Islands, Scotland          2987          4   
2                      Bourbon      Kentucky, USA            10          5   
3                      Bourbon      Kentucky, USA           628          4   
4           Peated Single Malt    Islay, Scotland            22          5   
...                        ...                ...           ...        ...   
3530          Flavored Whiskey      Kentucky, USA            13          1   
3531  Blended American Whiskey                USA           237          1   
3532      American Single Malt     Minnesota, USA             2          3   
3533             Other Whiskey        Oregon, USA             1          2   
3534  Blended American Whiskey      Missouri, USA            16          1   

        abv                                          cask_type  ...  oily  \
0     43.00  ex-bourbon American oak, ex-sherry European oa...  ...    20   
1     43.00                                          ex-sherry  ...    40   
2     57.10                          new, charred American oak  ...    15   
3     58.45                          new, charred American oak  ...    20   
4     53.90            ex-bourbon, ex-sherry, new Mizunara oak  ...    10   
...     ...                                                ...  ...   ...   
3530  43.00                          New, Charred American Oak  ...     0   
3531  40.00                                                oak  ...     0   
3532  42.00                          new, charred American oak  ...    40   
3533  44.50                          new, charred American oak  ...    80

In [14]:
print("Total Rating Dataset Size : {} ".format(processed_data['total_rating'].sum()))
print()
processed_data.info()

Total Rating Dataset Size : 907948 

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3535 entries, 0 to 3534
Data columns (total 24 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   link          3535 non-null   object 
 1   image         3535 non-null   object 
 2   name          3535 non-null   object 
 3   avr_rating    3464 non-null   float64
 4   category      3535 non-null   object 
 5   location      3535 non-null   object 
 6   total_rating  3535 non-null   int64  
 7   cost_rank     3535 non-null   int64  
 8   abv           3535 non-null   float64
 9   cask_type     2981 non-null   object 
 10  smoky         3535 non-null   int64  
 11  peaty         3535 non-null   int64  
 12  spicy         3535 non-null   int64  
 13  herbal        3535 non-null   int64  
 14  oily          3535 non-null   int64  
 15  full_bodied   3535 non-null   int64  
 16  rich          3535 non-null   int64  
 17  sweet         3535 non-null   int6

## 위스키 데이터 완성

In [16]:
savePath = os.getcwd()+"\\dataset"
processed_data.to_csv(savePath+"\\whisky.csv")